apps/backend/src/datasources/postgres/FapDataSource.ts

```diff
@@ -225,12 +238,16 @@ export default class PostgresFapDataSource implements FapDataSource {
         }
       })
       .then((allFaps: FapRecord[]) => {
-        const faps = allFaps.map((fap) => createFapObject(fap));
+        const faps = Promise.all(
+          allFaps.map(createFapObject).map(this.getSecretaries)
+        ).then((faps) => {
+          return {
+            totalCount: allFaps[0] ? allFaps[0].full_count : 0,
+            faps: faps,
+          };
+        });
 
-        return {
-          totalCount: allFaps[0] ? allFaps[0].full_count : 0,
-          faps,
-        };
+        return faps;
```

Comment:
I don't think there's a scoping issue, but this block feels a bit oversaturated with properties or variables that share the name `faps`

I think some of them could be eliminated to be a bit simpler, maybe start `return Promise.all(...` to elimintate one?